pkg/validator/validators.go

```diff
@@ -136,9 +160,52 @@ func ingressBackendValidator(req *admissionv1.AdmissionRequest) (*admissionv1.Ad
 		}
 	}
 
+	err := kc.checkDuplicateBackends(ingressBackend)
+	if errors.Is(err, errIngressBackendDuplicateBackends) {
+		return nil, err
+	}
+
+	// Fail open for transient errors, but log
+	if err != nil {
+		log.Error().Err(err).Msg("transient error encountered checking duplicate IngressBackend backends")
+	}
+
 	return nil, nil
 }
 
+// checkDuplicateBackends checks to see if the backends present in the given
+// ingressBackend conflict with existing ingressBackends
+func (kc *policyValidator) checkDuplicateBackends(candidateIngressBackend *policyv1alpha1.IngressBackend, existingIngressBackends ...policyv1alpha1.IngressBackend) error {
+	if candidateIngressBackend == nil {
+		return errors.New("candidateIngressBackend is nil")
+	}
+	ns := candidateIngressBackend.GetObjectMeta().GetNamespace()
+	if len(existingIngressBackends) == 0 {
+		ingressBackendList, err := kc.policyClient.PolicyV1alpha1().IngressBackends(ns).List(context.Background(), metav1.ListOptions{})
+		if err != nil {
+			return errors.Wrapf(err, "Error listing IngressBackend resources in namespace %s", ns)
+		}
+		existingIngressBackends = ingressBackendList.Items
+	}
+
+	var conflictString strings.Builder
+	for _, ib := range existingIngressBackends {
+		if conflicts := policy.DetectIngressBackendConflicts(*candidateIngressBackend, ib); len(conflicts) > 0 {
+			fmt.Fprintf(&conflictString, "[+] IngressBackend %s/%s conflicts with %s/%s:\n", ns, candidateIngressBackend.ObjectMeta.GetName(), ns, ib.ObjectMeta.GetName())
+			for _, err := range conflicts {
+				fmt.Fprintf(&conflictString, "%s\n", err)
+			}
+			fmt.Fprintf(&conflictString, "\n")
+		}
+	}
+
+	if conflictString.Len() != 0 {
+		return fmt.Errorf("%w\n%s", errIngressBackendDuplicateBackends, conflictString.String())
+	}
+
+	return nil
+}
+
```

Comment:
The logic here can be simplified by checking for duplicates inline while processing the backend. Can something like this be done:
When you see a backend, check the cache to see if the backend is already referenced in some other ingress resource. If it is, return an error. Otherwise, the backend is new.

Why do we need to match the entire IngressBackend object with all other IngressBackend objects?

Reply:
Fixed as we discussed offline